src/config.rs

```diff
@@ -393,6 +399,16 @@ impl Provider for Manager<'_> {
     fn get_current_user(&self) -> Option<&User> {
         self.config.users.iter().find(|user| user.current_user)
     }
+
+    fn set_last_used(&mut self) {
+        let mut user = self
+            .config
+            .users
+            .iter_mut()
+            .find(|user| user.current_user)
+            .unwrap();
+        user.last_used = Some(chrono::offset::Utc::now());
+    }
```

Comment:
What if the current user is overridden with the environment variables?

Reply:
Is there a way to know what user we are using? Perhaps `current_user` needs to take that into account

Reply:
I reckon we need to rethink the `config::Provider` interface. This is exposing how unfit for purpose it has become.